middlewares/pathReplace.go

```diff
@@ -0,0 +1,23 @@
+package middlewares
+
+import (
+	"net/http"
+)
+
+// PathReplace is a middleware used to replace the path of a URL request
+type PathReplace struct {
+	Handler http.Handler
+	Path    string
+}
+
+func (s *PathReplace) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	r.Header.Add("X-Replaced-Path", r.URL.Path)
+	r.URL.Path = s.Path
```

Comment:
Yes, the idea is to replace the entire path so that all paths will be forwarded to a single path on the backend.

For example, if I'm using Google Cloud Functions, I want both `http://example.com/` and `http://example.com/some/long/url.html` to be replaced with `/some-function`, then I can simulate having multiple paths on `https://some-project.cloudfunctions.net/some-function`.

Regarding your third question, I'm not sure why we would ever be unable to replace the path completely.